Storage/Storage/FaviconsTable.swift

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+private let TableNameFavicons = "favicons"
+
+// NOTE: If you add a new Table, make sure you update the version number in BrowserDB.swift!
+
+// This is our default favicons store.
+class FaviconsTable<T>: GenericTable<Favicon> {
+    private var files: FileAccessor
+    override var name: String { return TableNameFavicons }
+    override var rows: String { return "id INTEGER PRIMARY KEY AUTOINCREMENT, " +
+                       "url TEXT NOT NULL UNIQUE, " +
+                       "width INTEGER, " +
+                       "height INTEGER, " +
+                       "type INTEGER NOT NULL, " +
+                       "date REAL NOT NULL" }
+
+    init(files: FileAccessor) {
+        self.files = files
+    }
+
+    override func getInsertAndArgs(inout item: Favicon) -> (String, [AnyObject?])? {
```

Comment:
This is working around Swift compiler bugs. i.e. the wrong methods get called with out. There's a Note in Generic Table about the problem.